bundles/org.eclipse.cdt.lsp.clangd/OSGI-INF/org.eclipse.cdt.lsp.internal.clangd.ClangdConfigurationAccess.xml

```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.3.0" name="org.eclipse.cdt.lsp.internal.clangd.ClangdConfigurationAccess">
+   <service>
+      <provide interface="org.eclipse.cdt.lsp.clangd.ClangdConfiguration"/>
```

Comment:
What happens if a vendor adds his own provider for this service? Can we bound it to the used language server provider?

Reply:
This PR is focused at reworking access to preferences. We have a separate ticket #81 to work on configuration customization. I suggest to discuss the customization points and approaches there.
For example, we can add "server id" metadata to both "provider" and "configuration" and then invoke `ServiceCaller` with specific `filter` argument. Or think about something else.